src/code/audio/code_8019AF00.c

```diff
@@ -1953,17 +1957,153 @@ const char sAudioOcarinaUnusedText7[] = "check is over!!! %d %d %d\n";
+                randButtonIndex = Audio_NextRandom() % ARRAY_COUNT(sButtonToPitchMap);
+                sOcarinaSongNotes[OCARINA_SONG_TERMINA_WALL][j].pitch = sButtonToPitchMap[randButtonIndex];
+                sOcarinaSongNotes[OCARINA_SONG_TERMINA_WALL][j].length = 19;
+                sOcarinaSongNotes[OCARINA_SONG_TERMINA_WALL][j].volume = 80;
+                sOcarinaSongNotes[OCARINA_SONG_TERMINA_WALL][j].vibrato = 0;
+                sOcarinaSongNotes[OCARINA_SONG_TERMINA_WALL][j].bend = 0;
+                j++;
+                sOcarinaSongNotes[OCARINA_SONG_TERMINA_WALL][j].pitch = OCARINA_PITCH_NONE;
+                sOcarinaSongNotes[OCARINA_SONG_TERMINA_WALL][j].length = 3;
+                sOcarinaSongNotes[OCARINA_SONG_TERMINA_WALL][j].volume = 0;
+                sOcarinaSongNotes[OCARINA_SONG_TERMINA_WALL][j].vibrato = 0;
+                sOcarinaSongNotes[OCARINA_SONG_TERMINA_WALL][j].bend = 0;
+                j++;
+            }
+
+            sOcarinaSongNotes[OCARINA_SONG_TERMINA_WALL][j - 2].length = 90;
+            sOcarinaSongNotes[OCARINA_SONG_TERMINA_WALL][j - 1].length = 22;
+            sOcarinaSongNotes[OCARINA_SONG_TERMINA_WALL][j + 1].pitch = OCARINA_PITCH_NONE;
+            sOcarinaSongNotes[OCARINA_SONG_TERMINA_WALL][j + 1].length = 0;
+            AudioOcarina_MapSongFromNotesToButtons(OCARINA_SONG_TERMINA_WALL, OCARINA_SONG_TERMINA_WALL, 8);
+        } else {
+            j = Audio_NextRandom() % ARRAY_COUNT(sOoTOcarinaSongNotes);
+            ocarinaNote = sOoTOcarinaSongNotes[j];
+
+            for (; ocarinaNote[i].length != 0; i++) {
+                sOcarinaSongNotes[OCARINA_SONG_TERMINA_WALL][i] = ocarinaNote[i];
+            }
+
+            sOcarinaSongNotes[OCARINA_SONG_TERMINA_WALL][i].pitch = OCARINA_PITCH_NONE;
+            sOcarinaSongNotes[OCARINA_SONG_TERMINA_WALL][i].length = 0;
+            AudioOcarina_MapSongFromNotesToButtons(OCARINA_SONG_TERMINA_WALL, OCARINA_SONG_TERMINA_WALL,
+                                                   sOoTOcarinaSongsNumNotes[j]);
+        }
+    } while (AudioOcarina_TerminaWallValidateNotes(OCARINA_SONG_TERMINA_WALL, OCARINA_SONG_TERMINA_WALL) != 0);
+}
+
+/**
+ * Unused remnant of OoT
+ */
+void AudioOcarina_MemoryGameSetNumNotes(u8 minigameRound) {
+    u8 i;
+
+    if (minigameRound > 2) {
+        minigameRound = 2;
+    }
 
-#pragma GLOBAL_ASM("asm/non_matchings/code/code_8019AF00/func_8019D26C.s")
+    sOcarinaMemoryGameAppendPos = 0;
+    sOcaMemoryGameEndPos = sOcaMemoryGameNumNotes[minigameRound];
 
-#pragma GLOBAL_ASM("asm/non_matchings/code/code_8019AF00/func_8019D488.s")
+    for (i = 0; i < 3; i++) {
+        AudioOcarina_MemoryGameGenerateNotes();
+    }
+}
 
-#pragma GLOBAL_ASM("asm/non_matchings/code/code_8019AF00/func_8019D4F8.s")
+/**
+ * Unused remnant of OoT, Id 14 now represent Goron Lullaby Intro instead of the OoT ocarina memory game
+ */
+#define OCARINA_SONG_MEMORYGAME OCARINA_SONG_GORON_LULLABY_INTRO
```

Comment:
Kind of minor but I'd probably just define `OCARINA_SONG_MEMORYGAME` as 14 instead of `OCARINA_SONG_GORON_LULLABY_INTRO`, but keep the note about them sharing the ID. Main reason is that they aren't really connected, it just happens that 14 was chosen to be used for `OCARINA_SONG_GORON_LULLABY_INTRO`.

Reply:
I've gone ahead and taken this suggestion, plus I added a @bug comment about overwritting Goron Lullaby Intro if this function were to be called in MM